src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -258,29 +257,31 @@ parseModule = do
   exports <- P.optionMaybe $ parens $ commaSep1 parseDeclarationRef
   reserved "where"
   decls <- mark (P.many (same *> parseDeclaration))
+  _ <- P.eof
   end <- P.getPosition
   let ss = SourceSpan (P.sourceName start) (C.toSourcePos start) (C.toSourcePos end)
   return $ Module ss comments name decls exports
 
 -- | Parse a collection of modules in parallel
-parseModulesFromFiles :: forall m k. (MonadError MultipleErrors m) =>
-                                     (k -> FilePath) -> [(k, String)] -> m [(k, Module)]
-parseModulesFromFiles toFilePath input = do
-  modules <- flip parU id $ map wrapError $ inParallel $ flip map input $ \(k, content) -> do
+parseModulesFromFiles
+  :: forall m k
+   . MonadError MultipleErrors m
+  => (k -> FilePath)
+  -> [(k, String)]
+  -> m [(k, Module)]
+parseModulesFromFiles toFilePath input =
+  flip parU id $ map wrapError $ inParallel $ flip map input $ \(k, content) -> do
```

Comment:
Non-blocking: this can be simplified to `flip parU wrapError . inParallel . flip map input`.